Fixes installation instruction: sudo make install

flowejam opened the pull request:
Fixes a small issue which might cause a bit of friction for newcomers: containers/podman#24913

lsm5 reviewed (approved):
LGTM

lsm5 commented:
@TomSweeneyRedHat @rhatdan @DanBmh PTAL

TomSweeneyRedHat commented on getting-started/installation.md:
```diff
@@ -204,13 +204,17 @@ echo 'deb http://deb.debian.org/debian buster-backports main' | sudo tee -a /etc
 sudo apt-get -y -t buster-backports install libseccomp2
 sudo apt-get -y install podman
+# Restart dbus for rootless podman. Do this for every user using containers.
+# This command only works without root. One way to do this is login as the
```
```suggestion
# This command only works without root. One way to do this is to login as the
```

TomSweeneyRedHat commented:
One very small nit, but as it's user-facing documentation, I'd appreciate it if you could touch it up.  I'll merge it in once complete.

DanBmh commented:
Its now working for me. Not sure what did change on the pi or my login, I've seen that you didn't change the command.

lsm5 commented:
> Its now working for me. Not sure what did change on the pi or my login, I've seen that you didn't change the command.

wasn't the issue earlier that you were running `systemctl --user` as root?

DanBmh commented:
Yes, but as far as I remember I did try it without root too, and I was asked for some password, which often happens if I try to run some root only commands without sudo. So I thought I need sudo for this.

lsm5 commented:
> Yes, but as far as I remember I did try it without root too, and I was asked for some password, which often happens if I try to run some root only commands without sudo. So I thought I need sudo for this.

Ack, that sounds strange, but let us know if you see it again.

lsm5 reviewed (approved):
LGTM.

rhatdan commented:
Thanks @flowejam 
Please sign your PR
@Luap99 PTAL

Luap99 commented on docs/installation.md:
```diff
@@ -502,7 +502,7 @@ First, ensure that the `go version` that is found first on the $PATH is 1.16.x o
 cd podman
 make BUILDTAGS="selinux seccomp" PREFIX=/usr
-sudo make install PREFIX=/usr
+sudo env PATH=$PATH:/usr/local/go/bin make install PREFIX=/usr
```
I am bit confused, where go is depends how you installed it so this is just assuming how you in particular did install it. It must be in $PATH for both the make commands so showing it only on the sudo make install seems confusing.

Would it not be better to just write above that go must be in $PATH for all make commands?

hdub-tech replied:
(Not the OP, but someone who also ran into this issue and have been following along.) The installation instructions _do_ call out that Go needs to be on the PATH. However, when using sudo, `secure_path` is utilized (as OP mentions in containers/podman#24913) , and not the user's (or even root user's) PATH. I agree with OP that it is worth mentioning at least one possible solution, particularly since it is possible (likely?) the user needed to install Go from source in order to have a Podman compatible version.

That said, I think this could be simplified so there is no assumption about how or where Go is installed, while also avoiding the error that multiple people are running into. 

`sudo env PATH=$PATH make install PREFIX=/usr`

Instructions on the page already call out `export PATH=$GOPATH/bin:$PATH`, so this covers the install Go from source use case. And if a compatible version of Go was installed from a package manager, it will already be on the path. Demo from a "Go installed from source" environment:

```
$ which go
/usr/local/go/bin/go

$ sudo which go

$ sudo env PATH=$PATH which go
/usr/local/go/bin/go
```

flowejam replied:
@hdub-tech described the issue perfectly. I've incorporated your suggested changes @hdub-tech  - thanks!

Commit:
The env command is used to include the user's PATH environment variable.

Since the environment isn't preserved with `sudo make install`, `GO ?= go` in
the podman Makefile (commit: d3cd5098f0abb54279ce4abe594d309e2bc027b0)
will not have a value, causing the command to fail.

Fixes: https://github.com/containers/podman/issues/24913
Signed-off-by: James Flowers <bold.zone2373@fastmail.com>
Helped-by: H Dub <14808878+hdub-tech@users.noreply.github.com>

Luap99 reviewed (approved):
LGTM
